pymc3/step_methods/pgbart.py

```diff
@@ -56,223 +61,460 @@ class PGBART(ArrayStepShared):
     generates_stats = True
     stats_dtypes = [{"variable_inclusion": np.ndarray}]
 
-    def __init__(self, vars=None, num_particles=10, max_stages=5000, chunk="auto", model=None):
-        _log.warning("The BART model is experimental. Use with caution.")
+    def __init__(self, vars=None, num_particles=10, max_stages=100, chunk="auto", model=None):
+        _log.warning("BART is experimental. Use with caution.")
         model = modelcontext(model)
         initial_values = model.initial_point
-        vars = inputvars(vars)
-        self.bart = vars[0].distribution
+        value_bart = inputvars(vars)[0]
+        self.bart = model.values_to_rvs[value_bart].owner.op
+
+        self.X, self.Y, self.missing_data = preprocess_XY(self.bart.X, self.bart.Y)
+        self.m = self.bart.m
+        self.alpha = self.bart.alpha
+        self.k = self.bart.k
+        self.split_prior = self.bart.split_prior
+        if self.split_prior is None:
+            self.split_prior = np.ones(self.X.shape[1])
+
+        self.init_mean = self.Y.mean()
+        # if data is binary
+        Y_unique = np.unique(self.Y)
+        if Y_unique.size == 2 and np.all(Y_unique == [0, 1]):
```

Comment:
Is `self.Y` or `np.unique` sorted? Otherwise `[1, 0]` would be missed. 

Also, what about just checking if all unique values are discrete? That would also work for count data as you mention in the comment below...?

In either case, a test for discrete data would probably be good.

Reply:
It is sorted. 

My comment about count data is that we may need to check for count data and have some other rule to compute the std. I have some tests saying that this may not be actually necessary, still the comment is there as a reminder until I have more solid evidence.